### ISSUE TITLE
hot swaping test class & classloader cache

### ISSUE THREAD

**Issue by JigarJoshi** [testng-team/testng#492]
TLDR: It is not good to expose method that adds `classloader` in static cache using non static method

We  have a testconsole where we allow user to enter test class name and it loads that class using a new `URLClassLoader`

```
TestNG testNGRunner = new TestNG();
testNGRunner.setConfigFailurePolicy(XmlSuite.CONTINUE);
testNGRunner.setTestClasses(testClasses.toArray(new Class<?>[testClasses.size()]));
testNGRunner.setParallel(null);
testNGRunner.setThreadCount(1);
testNGRunner.run();
```

I found that TestNG again loads the class using its set of classloader in `TestNG.java` in `createCommandLineSuitesForMethods(List<String> commandLineMethods)`

```
Set<Class> classes = Sets.newHashSet();

for (String m : commandLineMethods) {
  classes.add(ClassHelper.forName(splitMethod(m)[0]));
}
```

This in turn calls `org.testng.internal.ClassHelper.forName(String)` which iterates through statically cached classloaders `org.testng.internal.ClassHelper.m_classLoaders` to load the class

This way it will never let me use new classloader that I am wishing it to use, I added a method in `ClassHelper.java` that resets its static classloading cache, that will not work for all but works for me because I have only one thread calling `TestNG` `run()`,

it could lead to thread synchronization issues if used in multi-thread caller environment

ideal solution would be to add capability in `TestNG` that let user choose what set of classloader they really want to use (global static ones or only those associated with state of this object)
